ios/src/platform_ios.mm

```diff
@@ -126,14 +132,115 @@ bool isContinuousRendering() {
     return ptr;
 }
 
-// No system fonts implementation (yet!)
-std::string systemFontPath(const std::string& _name, const std::string& _weight, const std::string& _face) {
-    return "";
+unsigned char* loadUIFont(UIFont* _font, size_t* _size) {
+
+    CGFontRef fontRef = CGFontCreateWithFontName((CFStringRef)_font.fontName);
+
+    if (!fontRef) {
+        *_size = 0;
+        return nullptr;
+    }
+
+    unsigned char* data = [CGFontConverter fontDataForCGFont:fontRef size:_size];
+
+    CGFontRelease(fontRef);
+
+    if (!data) {
+        LOG("CoreGraphics font failed to decode");
+
+        *_size = 0;
+        return nullptr;
+    }
+
+    return data;
+}
+
+std::vector<FontSourceHandle> systemFontFallbacksHandle() {
+    NSArray* fallbacks = [UIFont familyNames];
+
+    std::vector<FontSourceHandle> handles;
+
+    for (id fallback in fallbacks) {
+
+        UIFont* font = [UIFont fontWithName:fallback size:1.0];
+
+        size_t dataSize = 0;
+        auto cdata = loadUIFont(font, &dataSize);
+
+        if (!cdata) { continue; }
+
+        FontSourceHandle fontSourceHandle = [cdata, dataSize](size_t* _size) -> unsigned char* {
+            *_size = dataSize;
+
+            return cdata;
+        };
+
+        handles.push_back(fontSourceHandle);
+    }
+
+    return handles;
 }
 
-// No system fonts fallback implementation (yet!)
-std::string systemFontFallbackPath(int _importance, int _weightHint) {
-    return "";
+unsigned char* systemFont(const std::string& _name, const std::string& _weight, const std::string& _face, size_t* _size) {
+
+    static std::map<int, CGFloat> weightTraits = {
+        {100, UIFontWeightUltraLight},
+        {200, UIFontWeightThin},
+        {300, UIFontWeightLight},
+        {400, UIFontWeightRegular},
+        {500, UIFontWeightMedium},
+        {600, UIFontWeightSemibold},
+        {700, UIFontWeightBold},
+        {800, UIFontWeightHeavy},
+        {900, UIFontWeightBlack},
+    };
+
+    static std::map<std::string, UIFontDescriptorSymbolicTraits> fontTraits = {
+        {"italic", UIFontDescriptorTraitItalic},
+        {"oblique", UIFontDescriptorTraitItalic},
+        {"bold", UIFontDescriptorTraitBold},
+        {"expanded", UIFontDescriptorTraitExpanded},
+        {"condensed", UIFontDescriptorTraitCondensed},
+        {"monospace", UIFontDescriptorTraitMonoSpace},
+    };
+
+    UIFont* font = [UIFont fontWithName:[NSString stringWithUTF8String:_name.c_str()] size:1.0];
+
+    if (font == nil) {
+        // Get the default system font
+        if (_weight.empty()) {
+            font = [UIFont systemFontOfSize:0.0];
```

Comment:
You should make this `1.0` per the docs: https://developer.apple.com/reference/uikit/uifont/1619042-systemfontofsize?language=objc

"The size (in points) to which the font is scaled. This value must be greater than 0.0."